From: Ian Monroe <ian.monroe@gmail.com>
Date: Jul 18, 2006 12:00 PM
Subject: extragear/multimedia/amarok
To: kde-commits@kde.org
Cc: 130584-done@bugs.kde.org


SVN commit 563853 by ianmonroe:

 r5938@wasabi:  ian | 2006-07-18 11:57:06 -0500
 ProjectM works!
 Fix libvisual 0.4.0 support. Patch by Dennis Smit, thanks to Joe Rabinoff
 for the good bug report.
 BUG:130584


 _M            . (directory)
 M  +1 -0      ChangeLog
 M  +5 -6      src/vis/libvisual/libvisual.cpp


+++ trunk/extragear/multimedia/amarok/src/vis/libvisual/libvisual.cpp #563852:563853
@@ -301,13 +301,12 @@
     static int
     upload_callback( VisInput*, VisAudio *audio, void* )
     {
-        for( uint i = 0; i < 1024; i+=2 )
-        {
-            // amarok provides us with dual channel interleaved PCM
-            audio->plugpcm[0][i] = pcm_data[i];
-            audio->plugpcm[1][i] = pcm_data[i+1];
-        }
+        VisBuffer buf;

+        visual_buffer_init( &buf, pcm_data, 1024, 0 );
+        visual_audio_samplepool_input( audio->samplepool, &buf, VISUAL_AUDIO_SAMPLE_RATE_44100,
+            VISUAL_AUDIO_SAMPLE_FORMAT_S16, VISUAL_AUDIO_SAMPLE_CHANNEL_STEREO );
+
         return 0;
     }
